product-service/src/db.ts

```diff
@@ -0,0 +1,38 @@
+import * as productsList from './mock/products.json';
+
+export interface Product {
+    id: string;
+    imageURL: string;
+    price: string;
+}
+
+class DatabaseServiceMock {
+    products: Map<string, Product>;
+
+    constructor() {
+        this.products = new Map<string, Product>();
+        Array.from(productsList.keys()).forEach(productKey => {
```

Comment:
Possible. It looks a bit complicated. Do you think about complixety of this solution? Just:
private products = productsList;
getProducts(): Product[] {
return this.products;
}
getProductsById(id: string):Product {
  const product = this.products.find((item:Product) => item.id === id);
  ...
}

Reply:
Although this code looks confusing, I want to use a `map`  for the underlying data structures.
I would like to keep it this way if it doesn't affect the grading. ;)

Reply:
@numatay Of course, it doesn't affect the grading